test/extension/handlers/lifecycle/handlers.go

```diff
@@ -0,0 +1,180 @@
+/*
+Copyright 2022 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// Package lifecycle contains the handlers for the lifecycle hooks.
+package lifecycle
+
+import (
+	"context"
+	"fmt"
+	"strconv"
+
+	"github.com/pkg/errors"
+	"gopkg.in/yaml.v2"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/types"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	runtimehooksv1 "sigs.k8s.io/cluster-api/exp/runtime/hooks/api/v1alpha1"
+	runtimecatalog "sigs.k8s.io/cluster-api/internal/runtime/catalog"
+)
+
+// Handler is the handler for the lifecycle hooks.
+type Handler struct {
+	Client client.Client
+}
+
+// DoBeforeClusterCreate implements the BeforeClusterCreate hook.
+func (h *Handler) DoBeforeClusterCreate(ctx context.Context, request *runtimehooksv1.BeforeClusterCreateRequest, response *runtimehooksv1.BeforeClusterCreateResponse) {
+	log := ctrl.LoggerFrom(ctx)
+	log.Info("BeforeClusterCreate is called")
+	cluster := request.Cluster
+	if err := h.recordCallInConfigMap(cluster.Name, cluster.Namespace, runtimehooksv1.BeforeClusterCreate); err != nil {
+		response.Status = runtimehooksv1.ResponseStatusFailure
+		response.Message = err.Error()
+		return
+	}
+	log.Info("BeforeClusterCreate has been recorded in configmap", "cm", cluster.Name+"-hookresponses")
+
+	err := h.readResponseFromConfigMap(cluster.Name, cluster.Namespace, runtimehooksv1.BeforeClusterCreate, response)
+	if err != nil {
+		response.Status = runtimehooksv1.ResponseStatusFailure
+		response.Message = err.Error()
+		return
+	}
+
+	log.Info(fmt.Sprintf("BeforeClusterCreate responding RetryAfterSeconds: %v\n", response.RetryAfterSeconds))
```

Comment:
```suggestion
	log.Info(fmt.Sprintf("BeforeClusterCreate responding RetryAfterSeconds: %d\n", response.RetryAfterSeconds))
```
nit: let's use %d

Reply:
Should we add a log like this to all handlers? WDYT about adding something with defer in l.45 which prints the entire response nicely formatted?